server/src/crm/crm.utilities.ts

```diff
@@ -109,5 +98,79 @@ export function containsAnyOf(propertyName, strings = [], options?) {
     .join(' or ');
   const lambdaQueryPrefix = childEntity ? `${childEntity}/any` : '';
 
-  return `(${not ? 'not ': ''}${lambdaQueryPrefix}(${containsQuery}))`;
+  return `(${not ? 'not ' : ''}${lambdaQueryPrefix}(${containsQuery}))`;
+}
+
+export const dateParser = function (key, value) {
+  if (typeof value === 'string') {
+    // YYYY-MM-DDTHH:mm:ss.sssZ => parsed as UTC
+    // YYYY-MM-DD => parsed as local date
+
+    if (value != "") {
+      const a = /^(\d{4})-(\d{2})-(\d{2})T(\d{2}):(\d{2}):(\d{2}(?:\.\d*)?)Z$/.exec(value);
+
+      if (a) {
+        const s = parseInt(a[6]);
+        const ms = Number(a[6]) * 1000 - s * 1000;
+        return new Date(Date.UTC(parseInt(a[1]), parseInt(a[2]) - 1, parseInt(a[3]), parseInt(a[4]), parseInt(a[5]), s, ms));
+      }
+
+      const b = /^(\d{4})-(\d{2})-(\d{2})$/.exec(value);
+
+      if (b) {
+        return new Date(parseInt(b[1]), parseInt(b[2]) - 1, parseInt(b[3]), 0, 0, 0, 0);
+      }
+    }
+  }
+
+  return value;
+}
```

Comment:
This is used in crm.service and does the same thing as the old _dateReviver method from the old crm module.

### Why

This detects Date-like strings when parsing JSON. There is no Date-type in JSON. So, when parsing JSON, this tries to help by converting those to JavaScritp Date Objects. Upstream, we have code that depends on this, so I've kept the behavior.

Reply:
Thank you